x-pack/plugins/observability/server/domain/services/composite_slo/validate_composite_slo.test.ts

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  createWeightedAverageSource,
+  createCompositeSLO,
+} from '../../../services/composite_slo/fixtures/composite_slo';
+import { createSLO } from '../../../services/slo/fixtures/slo';
+import {
+  sevenDaysRolling,
+  thirtyDaysRolling,
+  weeklyCalendarAligned,
+} from '../../../services/slo/fixtures/time_window';
+import { validateCompositeSLO } from '.';
+
+describe('validateCompositeSLO', () => {
+  it("throws when specified combined SLOs don't match the actual SLO revision", () => {
+    const sloOne = createSLO({ revision: 3 });
+    const sloTwo = createSLO({ revision: 2 });
+    const compositeSlo = createCompositeSLO({
+      sources: [
+        createWeightedAverageSource({ id: sloOne.id, revision: sloOne.revision }),
+        createWeightedAverageSource({ id: sloTwo.id, revision: 1 }),
+      ],
+    });
+    expect(() => validateCompositeSLO(compositeSlo, [sloOne, sloTwo])).toThrowError(
+      'One or many source SLOs are not matching the specified id and revision.'
+    );
+  });
+
+  it('throws when specified combined SLOs refers to a non-existant SLO', () => {
+    const sloOne = createSLO({ revision: 3 });
+    const compositeSlo = createCompositeSLO({
+      sources: [
+        createWeightedAverageSource({ id: sloOne.id, revision: sloOne.revision }),
+        createWeightedAverageSource({ id: 'non-existant' }),
+      ],
+    });
+    expect(() => validateCompositeSLO(compositeSlo, [sloOne])).toThrowError(
+      'One or many source SLOs are not matching the specified id and revision.'
+    );
+  });
+
+  it('throws when the time window is not the same accros all combined SLOs', () => {
```

Comment:
Is the difference between this test and the next one the type of time window (rolling vs. calendar) and both rolling but different ranges (seven, thirty)?

Reply:
Yes, this test checks the type of the time window (calendar aligned vs rolling), the next one checks the duration of a time window when all are rolling.